regolith/tools.py

```diff
@@ -1282,9 +1245,82 @@ def fragment_retrieval(coll, fields, fragment, case_sensitive = False):
                     break
     return ret_list
 
+
 def get_id_from_name(coll, name):
     person = fuzzy_retrieval(coll,["name", "aka", "_id"], name,
                              case_sensitive=False)
     if person:
         return person["_id"]
-    else: return None
\ No newline at end of file
+    else: return None
+
+
+def is_fully_appointed(person, begin_date=None, end_date=None):
+    """Checks if a collection of appointments for a person is valid and fully loaded
+        for a given interval of time
+
+        Parameters
+        ----------
+        person: dict
+            The person whose appointments need to be checked
+        begin_date: datetime, string, optional
+            The start date of the interval of time to check appointments for
+        end_date: datetime, string, optional
+            The end date of the interval of time to check appointments for
+
+        Returns
+        -------
+        bool:
+            True if the person is fully appointed and False if not
+
+        Examples
+        --------
+        >>> appts = [{"begin_year": 2017, "begin_month": 6, "begin_day": 1, "end_year": 2017, "end_month": 6,\
+         "end_day": 15, "grant": "grant1", "loading": 1.0, "type": "pd", }, {"begin_year": 2017, "begin_month": 6, \
+         "begin_day": 17,  "end_year": 2017,  "end_month": 6, "end_day": 30, "grant": "grant2", "loading": 1.0, \
+         "type": "pd",} ]
+        >>> aejaz = {"name": "Adiba Ejaz", "_id": "aejaz", "appointments": appts}
+        >>> is_fully_appointed(aejaz, "2017-06-01", "2017-06-30")
+
+        In this case, we have an insufficient loading and minimum of 0.0 on 2017-06-16, hence it would return False and
+        print "aejaz is short of full loading with minimim 0.0 on 2017-06-16".
+        """
+
+    status = True
+    appts = person.get('appointments')
+    earliest = get_dates(appts[0])['begin_date']
```

Comment:
a cleaner way of doing this might be something like:
```
dates = [(get_dates(appt).get('begin_date'), get_dates(appt).get('end_date'))  for appt in appts]
if not begin_date:
    begin_date = min(dates[0])
if not end_date:
    latest = max(dates[1])
```
However, this looks to me like "magic" that we should remove it.  begin and end date should be required arguments and the function should blow up if they are not given.   The "no magic" rule.

Reply:
Thank you for the logic, it's very elegant! I will keep it in mind for future code, and change these arguments to required

Reply:
both should be required, just as you did in the code.  Remember, no magic.   You can spend half a lifetime looking for bugs generated by magic.